Lesson 2/app/src/main/java/com/mobileapps/lesson2/LoginFragment.kt

```diff
@@ -0,0 +1,57 @@
+package com.mobileapps.lesson2
+
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import com.google.android.material.button.MaterialButton
+import com.google.android.material.textfield.TextInputLayout
+
+
+class LoginFragment: Fragment() {
+
+    private lateinit var emailTextInputLayout: TextInputLayout
+    private lateinit var passwordTextInputLayout: TextInputLayout
+    private lateinit var nextButton: MaterialButton
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_login, container, false)
+    }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        emailTextInputLayout = view.findViewById(R.id.emailTextInputLayout)
+        passwordTextInputLayout = view.findViewById(R.id.passwordTextInputLayout)
+        nextButton = view.findViewById(R.id.nextButton)
+        val registerNowTextView = view.findViewById<TextView>(R.id.registerNowTextView)
+
+        registerNowTextView.setOnClickListener {
+            (activity as? AccountActivity)?.replaceFragment(RegisterFragment())
```

Comment:
Don't ever cast fragment's parent activity to a specific one. Use interfaces instead. Fragments are meant to be reusable pieces of UI - if you do it like that, it's firmly tied to that very activity + if somebody tries to use it in a different activity - he won't even know that part of fragment's logic won't work, because it won't crash on fragment's onAttach